kolibri/core/content/utils/channel_import.py

```diff
@@ -598,7 +599,14 @@ def postgres_table_import(self, model, row_mapper, table_mapper):
         def generate_data_with_default(record):
             for col_name, column_obj in columns:
                 default = self.get_and_set_column_default(column_obj)
-                value = row_mapper(record, column_obj.name)
+                value = row_mapper(record, col_name)
+                if not column_obj.primary_key and isinstance(
+                    column_obj.type, sa_String
+                ):
+                    max_length = column_obj.type.length
+                    if max_length is not None:
+                        value = value[:max_length] if value is not None else default
```

Comment:
Is this to handle cases where even with proper handling of utf-8 the tag name is too long because of slack of checks in generation of the SQLite db?

Reply:
yes,  while testing this, I've seen tag names 40 chars long in sqlite while max_length is 30 in the model.